components/planner/history/StudentHistoryView/CourseAudit.tsx

```diff
@@ -0,0 +1,686 @@
+                onChange={handleChange}
+              ></TextField>
+              <FormControl className={classes.wider}>
+                <InputLabel shrink={true} id="year">
+                  Year
+                </InputLabel>
+                <Select labelId="year" id="year" value={year} onChange={handleChange} name={'Year'}>
+                  {returnOptions(returnYears())}
+                </Select>
+              </FormControl>
+              <FormControl className={classes.wider}>
+                <InputLabel shrink={true} id="season">
+                  Season
+                </InputLabel>
+                <Select
+                  labelId="season"
+                  id="season"
+                  value={season}
+                  onChange={handleChange}
+                  name={'Season'}
+                >
+                  {returnOptions(Object.keys(SEMESTERS))}
+                </Select>
+              </FormControl>
+              <FormControl className={classes.wider}>
+                <InputLabel shrink={true} id="grade">
+                  Grade
+                </InputLabel>
+                <Select
+                  labelId="grade"
+                  id="grade"
+                  value={grade}
+                  onChange={handleChange}
+                  name={'Grade'}
+                >
+                  {returnOptions(Object.keys(LETTERS))}
+                </Select>
+              </FormControl>
+            </div>
+          );
+        } else if (selectedState.length > 0) {
+          return (
+            <Typography
+              className={classes.title}
+              color="inherit"
+              variant="subtitle1"
+              component="div"
+            >
+              {selectedState.length} selected
+            </Typography>
+          );
+        } else {
+          return (
+            <Typography className={classes.title} variant="h6" id="tableTitle" component="div">
+              Course History
+            </Typography>
+          );
+        }
+      })()}
+      {(() => {
+        if (adding) {
+          return (
+            <Tooltip title="Done">
+              <IconButton aria-label="done" onClick={addStuff}>
+                <CheckIcon />
+              </IconButton>
+            </Tooltip>
+          );
+        } else if (selectedState.length > 0) {
+          return (
+            <Tooltip title="Delete Course">
+              <IconButton aria-label="delete" onClick={deleteStuff}>
+                <DeleteIcon />
+              </IconButton>
+            </Tooltip>
+          );
+        } else {
+          return (
+            <Tooltip title="Add Course">
+              <IconButton aria-label="add" onClick={addStuff}>
+                <AddIcon />
+              </IconButton>
+            </Tooltip>
+          );
+        }
+      })()}
+    </Toolbar>
+  );
+};
+
+//Colors the grade according to its GPA value
+//Calculates based on a rational function with a constant
```

Comment:
Use doc comments, and could you be more clear what "based on a rational function with a constant" means?
```suggestion
/**
 * Colors the grade according to its GPA value.
 */
```